router/router.go

```diff
@@ -752,52 +749,56 @@ func (worker *workerT) handleWorkerDestinationJobs(ctx context.Context) {
 					}
 					respStatusCode, respBody = worker.rt.customDestinationManager.SendData(destinationJob.Message, sourceID, destinationID)
 				} else {
-					result := getIterableStruct(destinationJob.Message, transformAt)
-					for _, val := range result {
-						err := integrations.ValidatePostInfo(val)
-						if err != nil {
-							respStatusCode, respBodyTemp = 400, fmt.Sprintf(`400 GetPostInfoFailed with error: %s`, err.Error())
-							respBodyArr = append(respBodyArr, respBodyTemp)
-						} else {
-							// stat start
-							pkgLogger.Debugf(`responseTransform status :%v, %s`, worker.rt.transformerProxy, worker.rt.destName)
-							sendCtx, cancel := context.WithTimeout(ctx, worker.rt.netClientTimeout)
-							defer cancel()
-							//transformer proxy start
-							if worker.rt.transformerProxy {
-								rtl_time := time.Now()
-								respStatusCode, respBodyTemp = worker.rt.transformer.ProxyRequest(ctx, val, worker.rt.destName)
-								worker.routerProxyStat.SendTiming(time.Since(rtl_time))
-								authType := router_utils.GetAuthType(destinationJob.Destination)
-								if router_utils.IsNotEmptyString(authType) && authType == "OAuth" {
-									pkgLogger.Debugf(`Sending for OAuth destination`)
-									// Token from header of the request
-									respStatusCode, respBodyTemp = worker.rt.HandleOAuthDestResponse(&HandleDestOAuthRespParamsT{
-										ctx:            ctx,
-										destinationJob: destinationJob,
-										workerId:       worker.workerID,
-										trRespStCd:     respStatusCode,
-										trRespBody:     respBodyTemp,
-										secret:         destinationJob.JobMetadataArray[0].Secret,
-									})
-								}
-							} else {
-								rdl_time := time.Now()
-								resp := worker.rt.netHandle.SendPost(sendCtx, val)
-								respStatusCode, respBodyTemp, respContentType = resp.StatusCode, string(resp.ResponseBody), resp.ResponseContentType
-								// stat end
-								worker.routerDeliveryLatencyStat.SendTiming(time.Since(rdl_time))
-							}
-							// transformer proxy end
-							if isSuccessStatus(respStatusCode) {
+					result, err := getIterableStruct(destinationJob.Message, transformAt)
+					if err != nil {
+						respStatusCode, respBody = 400, fmt.Errorf("transformer response unmarshal error: %w", err).Error()
```

Comment:
Failure to process the server's response is actually a client processing error, however we are masking it as a server error `400` response. This appears to be a pattern that we are already using, so it is acceptable.
Nevertheless, would it be valuable to discriminate client processing errors by using some special status code?

The following link provides all assigned (& unassigned) status codes:
https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
Ranges 419-420, 432-450 & 452-499 are unassigned. I would pick 499 which might be the last status code we should expect to become assigned to some special meaning.